taipy/core/_entity/_ready_to_run_property.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021-2024 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from typing import TYPE_CHECKING, Dict, Set, Union
+
+from ..notification import EventOperation, Notifier, _make_event
+
+if TYPE_CHECKING:
+    from ..data.data_node import DataNode, DataNodeId
+    from ..scenario.scenario import Scenario, ScenarioId
+    from ..sequence.sequence import Sequence, SequenceId
+    from ..task.task import Task, TaskId
+
+
+class _ReadyToRunProperty:
+    IS_SUBMITTABLE_PROPERTY_NAME: str = "is_submittable"
+
+    # A dictionary of the data nodes not ready_to_read and their corresponding submittable entities.
+    _datanode_id_submittables: Dict["DataNodeId", Set[Union["ScenarioId", "SequenceId", "TaskId"]]] = {}
+
+    # A nested dictionary of the submittable entities (Scenario, Sequence, Task) and
+    # the data nodes that make it not ready_to_run with the reason(s)
+    _submittable_id_datanodes: Dict[Union["ScenarioId", "SequenceId", "TaskId"], Dict["DataNodeId", Set[str]]] = {}
+
+    @classmethod
+    def _add(cls, dn: "DataNode", reason: str) -> None:
+        from ..scenario.scenario import Scenario
+        from ..sequence.sequence import Sequence
+        from ..task.task import Task
```

Comment:
If we can
```python
parent_entities.get("scenario", []
```
instead of
```python
parent_entities.get(Scenario._MANAGER_NAME, []
```
we probably don't need these import lines. However, the practice of using _MANAGER_NAME is ok to me as well

Reply:
I'll keep the import for manager_name instead to avoid hard-coding things :D